### PR TITLE
KAA-178: Create a unified plugin system for log appenders

### PR DESCRIPTION
**Pull request by ikulikov**

### CODE REVIEW

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/persistence/KaaClientPropertiesState.java [hunk 1]
```diff
@@ -108,7 +108,9 @@ public KaaClientPropertiesState(PersistentStorage storage, KaaClientProperties p
         state = new Properties();
         if (storage.exists(stateFileLocation)) {
             try {
-                state.load(storage.openForRead(stateFileLocation));
+                InputStream stream = storage.openForRead(stateFileLocation);
+                state.load(stream);
+                stream.close();
```
Comment (ashvayka) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/persistence/KaaClientPropertiesState.java]: This should be moved to finally block

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/persistence/KaaClientPropertiesState.java [hunk 2]
```diff
@@ -230,6 +232,7 @@ public PrivateKey getPrivateKey() {
             try {
                 InputStream input = storage.openForRead(clientPrivateKeyFileLocation);
                 privateKey = KeyUtil.getPrivate(input);
+                input.close();
```
Comment (ashvayka) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/persistence/KaaClientPropertiesState.java]: This should be moved to finally block

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/persistence/KaaClientPropertiesState.java [hunk 3]
```diff
@@ -259,6 +262,7 @@ public PublicKey getPublicKey() {
             try {
                 InputStream input = storage.openForRead(clientPublicKeyFileLocation);
                 publicKey = KeyUtil.getPublic(input);
+                input.close();
```
Comment (ashvayka) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/persistence/KaaClientPropertiesState.java]: This should be moved to finally block

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/persistence/KaaClientPropertiesState.java [hunk 4]
```diff
@@ -230,6 +232,7 @@ public PrivateKey getPrivateKey() {
             try {
                 InputStream input = storage.openForRead(clientPrivateKeyFileLocation);
                 privateKey = KeyUtil.getPrivate(input);
```
Comment (shpigor) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/persistence/KaaClientPropertiesState.java:242]: InputStream will be closed in KeyUtil.getPrivate(input) method.

**Comment by ashvayka** [comment]
some minor comments about finally, otherwise LGTM

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/persistence/KaaClientPropertiesState.java [hunk 5]
```diff
@@ -259,6 +262,7 @@ public PublicKey getPublicKey() {
             try {
                 InputStream input = storage.openForRead(clientPublicKeyFileLocation);
                 publicKey = KeyUtil.getPublic(input);
```
Comment (shpigor) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/persistence/KaaClientPropertiesState.java:281]: InputStream will be closed in KeyUtil.getPublic(input) method.

File: server/appenders/oracle-nosql/appender/pom.xml [hunk 6]
```diff
@@ -40,10 +40,37 @@
...
+        </dependency>
+        <dependency>
         	<groupId>com.oracle.kv</groupId>
         	<artifactId>oracle-nosql-client</artifactId>
         	<version>3.0.14</version>
         </dependency>
+        <dependency>
+            <groupId>com.oracle.kv</groupId>
+            <artifactId>kvstore</artifactId>
+            <version>3.1.7</version>
```
Comment (shpigor) [inline server/appenders/oracle-nosql/appender/pom.xml]: Could you move dependencies versions to properties or parent pom?

**Comment by shpigor** [comment]
Reviewed

**Pull request merged**